core/src/mender-client.c

```diff
@@ -1007,7 +1011,11 @@ mender_client_update_work_function(void) {
             case MENDER_UPDATE_STATE_REBOOT:
                 assert(mender_update_module->requires_reboot);
                 mender_log_info("Artifact installation done, rebooting");
-                mender_client_publish_deployment_status(deployment_id, MENDER_DEPLOYMENT_STATUS_REBOOTING);
+                if (MENDER_ABORTED == mender_client_publish_deployment_status(deployment_id, MENDER_DEPLOYMENT_STATUS_REBOOTING)) {
+                    update_state = MENDER_UPDATE_STATE_FAILURE;
+                    set_and_store_state(update_state);
+                    continue;
```

Comment:
Why not leave `ret` and `NEXT_STATE` handle this just like in the `INSTALL` state?

Reply:
Well, reboot goes to `MENDER_UPDATE_STATE_ROLLBACK` on failure, do we really want to go there?

Reply:
Why not? Shouldn't we roll back the aborted deployment?

Reply:
We haven't rebootet yet at this stage, though. Won't going to rollback end up in calling the reboot callback, which will be wrong, since we haven't actually switched to the other partition?

Reply:
My thinking is that an aborted deployment should be like any other failure. So here, in particular, I don't see why it should be treated differently (state-wise) from a failure of the update module's callback for the `REBOOT` state. Any exception to the success/failure state transitions must be justified as absolutely necessary. If following the "failure path" here can cause issues, we should fix the failure path, I believe.

Reply:
Yeah, when I think about it, it does indeed make sense to deviate as little as possible from the normal flow, thanks for your reasoning